flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/State.java

```diff
@@ -51,7 +51,8 @@ default void onLeave(Class<? extends State> newState) {}
     void suspend(Throwable cause);
 
     /**
-     * Gets the current {@link JobStatus}.
+     * Gets the current {@link JobStatus}. The returned job status must remain valid for as long as
+     * the scheduler is in this state.
```

Comment:
I wasn't exactly sure about what this comment means at first. What do you think about rephrasing it a little?

```
The returned job status will remain unchanged until the scheduler transitions to a different state.
```

Reply:
I struggled with the phrasing a bit, yours is better 👍